models/Asset.php

```diff
@@ -627,8 +627,12 @@ public function correctPath()
                 throw new Exception("invalid filename '" . $this->getKey() . "' for asset with id [ " . $this->getId() . ' ]');
             }
 
+            if($this->getParentId() === null || $this->getParentId() === 0) {
```

Comment:
```suggestion
            if(!$this->getParentId()) {
```
Should be sufficient here 😊